### PR TITLE
Introduce CI workflow

### PR DESCRIPTION
**Pull request by hangingman**
ref #68

### Introduce GitHub Action, CI workflow.
- This will test;
    - ubuntu(focal): jdk8, jdk11
    - debian(bullseye): jdk11

This workflow is using a dockr image, `dpokidov/imagemagick:7.1.1-10-{ubuntu,bullseye}` . The docker image accelerate compile time because the docker image contains prebuild ImageMagick.

### configure.ac and some script improvement

CHANGES;

- configure.ac cosmetic changes
    - Use AC_SYS_LARGEFILE instead of own checks
    - Use AS_HELP_STRING, to arrange help string indent

- configure.ac functional changes
    - Add `--with-magickwand-config` option, it enable users to pass their own magickwand-config path
    - Replace `AC_PROG_LIBTOOL` to `LT_INIT`

- automake changes
    - Remove `@OS@` from `Make.def.in`, because it's only used by `ifeq ($(strip $(OS)),Darwin)`
    - Change `CYGPATH` as automake macro using `AM_CONDITIONAL`

- misc
    - Add .gitignore entries
    - Remove .travis.yml

## test

- following build environments are guaranteed successful build by CI
    - [x] debian11
    - [x] ubuntu(focal)

- On mac, I found build error. But it seems caused by imagemagick m4 macro. Currently, I couldn't resolve.
    * Mac(monterey)
    * ref: https://github.com/ImageMagick/ImageMagick/issues/1073

```shell
// install imagemagick--7.1.1-12
$ brew install imagemagick
$ brew install openjdk@11
$ sudo ln -sfn /usr/local/opt/openjdk@11/libexec/openjdk.jdk /Library/Java/JavaVirtualMachines/openjdk-11.jdk
$ ./configure --with-java-home=/Library/Java/JavaVirtualMachines/openjdk-11.jdk/Contents/Home
...
clang: error: unsupported option '-fopenmp'
...
```

## TODO
- Need to set-up a branch protection, which is disable GitHub pull request merge button when CI job got failed
    - https://github.com/orgs/community/discussions/21611

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
-RUN apt-get install -y openjdk-7-jdk libmagickcore-dev libmagickwand-dev make
+RUN apt-get update -y
+RUN apt-get install -y autoconf automake libtool build-essential pkg-config openjdk-11-jdk
 
-ADD docker/build.sh /
-RUN chmod +x /build.sh
-WORKDIR /src
-VOLUME /src
-VOLUME /build
-CMD /build.sh
```
Comment (hangingman) [inline Dockerfile:11]: I should have left `VOLUME /build` ? This was probably used for retrieving compiled binary.

**Comment by hangingman** [comment]
@nordfalk 
Hi, I created a pull request as I mentioned in PR #66
Please take a look

**Comment by nordfalk** [comment]
Hi @hangingman, it looks fine to me, but I didnt tried to build it.
Is it ready to merge? 

@Waterpicker or @MolotovCherry or @a-schild , what do you think of this PR?

**Comment by hangingman** [comment]
@nordfalk 
Thank you.
It's okay to merge.

During this work, I found a build error on Mac. If it's necessary, I will create an issue (but, who cares it ? 🤔 )

File: .gitignore [hunk 2]
```diff
@@ -11,6 +11,17 @@ obj
...
 /config.status
 /configure
+/configure~
 /ltmain.sh
 /config.guess
 /config.sub
+/missing
+/Makefile
+/Makefile.in
+/NEWS
```
Comment (nordfalk) [inline .gitignore:21]: NEWS, AUTHORS, COPYING, ChangeLog shouldnt be in .gitignore.

**Review by nordfalk — APPROVED** [review]
You might want to add a note to ChangeLog.txt.
Perhaps just
"See https://github.com/techblue/jmagick/commits/master for recent changes"

File: .gitignore (hunk 2 above)
Comment (hangingman) [reply .gitignore]: Fixed

**Pull request merged**